docs/core/concepts/secrets.md

```diff
@@ -0,0 +1,42 @@
+# Secrets
+
+## Overview
+
+Very often, workflows require sensitive information to run: API keys, passwords, tokens, credentials, etc. As a matter of best practice, such information should never be hardcoded into a workflow's source code, as the code itself will need to be guarded. Furthermore, sensitive information should not be provided via a Prefect `Parameter`, because Parameters, like many tasks, can have their results stored.
+
+Prefect provides a mechanism called `Secrets` for working with sensitive information. `Secrets` are a special kind of task; they may be instantiated and used as the input to any other task. However, `Secret` tasks differ from regular tasks in two ways:
```

Comment:
I think you might want to have two sections here:
- one for secret tasks, as you already have
- one for Secrets (the non-task version), which are what you'd need inside result handlers / state handlers / notification callbacks